goodwe/et.py

```diff
@@ -40,14 +41,21 @@ class ET(Inverter):
                    max(0, read_bytes4(data, 35113)) +
                    max(0, read_bytes4(data, 35117)),
                    "PV Power", "W", Kind.PV),
-        ByteH("pv4_mode", 35119, "PV4 Mode code", "", Kind.PV),  # l
-        EnumL("pv4_mode_label", 35119, PV_MODES, "PV4 Mode", Kind.PV),
-        ByteH("pv3_mode", 35119, "PV3 Mode code", "", Kind.PV),  # h
-        EnumH("pv3_mode_label", 35119, PV_MODES, "PV3 Mode", Kind.PV),
-        ByteL("pv2_mode", 35120, "PV2 Mode code", "", Kind.PV),  # l
-        EnumL("pv2_mode_label", 35120, PV_MODES, "PV2 Mode", Kind.PV),
-        ByteH("pv1_mode", 35120, "PV1 Mode code", "", Kind.PV),  # h
-        EnumH("pv1_mode_label", 35120, PV_MODES, "PV1 Mode", Kind.PV),
+# =============================================================================
+# Changed gowima: Byte order in the implementation of ByteH|L and EnumH|L
+#                 is big endian, which is in accordance to the GoodWe Modbus 
+#                 register documentation. So the byte sequence left to right 
+#                 in a register is H - L. Modes are stored as a byte sequence
+#                 pv4_mode ... pv1_mode. ---> ByteH/ByteL have to be exchanged 
+#                 pairwase for mode 4, 3 and 2, 1.
```

Comment:
There was conflict with another PR merged today, so I took the liberty and took this fix and committed it directly.
Please rebase/resolve the conflict, I will review the rest tomorrow.
And thanks for noticing and fixing this !